update serviceaccount cmds to use externals

juanvallejo opened the pull request:
cc @deads2k @soltysh

juanvallejo commented:
cc @deads2k

deads2k commented on pkg/serviceaccounts/util/helpers.go:
```diff
@@ -14,26 +14,56 @@ const (
-	return apihelpers.GetName(serviceAccount.Name, "dockercfg-", maxSecretPrefixNameLength)
+// IsValidServiceAccountToken returns true if the given secret contains a service account token valid for the given service account
+// TODO, these should collapse onto the upstream method
+func IsValidServiceAccountToken(serviceAccount *kapi.ServiceAccount, secret *kapi.Secret) bool {
```
This just pushed your problem around.  It didn't fix it.  Leave this one where it lies and make your external one.

deads2k commented on pkg/serviceaccounts/util/helpers.go:
```diff
@@ -1,7 +1,7 @@
 import (
-	"k8s.io/api/core/v1"
+	corev1 "k8s.io/api/core/v1"
 	kapi "k8s.io/kubernetes/pkg/apis/core"
```
this dependency needs to go.

juanvallejo commented:
@deads2k thanks, review comments addressed

deads2k commented:
And now I've looked at usage. Looks like disjoint sets of helpers. One set for `oc` (move to `oc/lib`) and one set for a single controller (move to point of use) and one also used by test/util (switch to upstream).

juanvallejo commented:
@deads2k 
> And now I've looked at usage. Looks like disjoint sets of helpers. One set for oc (move to oc/lib) and one set for a single controller (move to point of use) and one also used by test/util (switch to upstream).

Thanks, created oc/lib/serviceaccount for use with oc commands, moved controller-specific helpers to point of use, and switched that test to use the upstream helper

juanvallejo commented:
/test gcp

juanvallejo commented:
@soltysh friendly ping

juanvallejo commented:
/test gcp

juanvallejo commented:
/test gcp

soltysh commented:
/retest

soltysh commented:
/retest

juanvallejo commented:
/test gcp

juanvallejo commented:
/retest

juanvallejo commented:
/test gcp
